cloudinit/sources/DataSourceOracle.py

```diff
@@ -317,11 +323,19 @@ def _add_network_config_from_opc_imds(self, set_primary: bool = False):
                 )
             elif self._network_config["version"] == 2:
                 self._network_config["ethernets"][name] = {
-                    "addresses": [f"{vnic_dict['privateIp']}/{prefix}"],
+                    "addresses": [
+                        f"{vnic_dict['privateIp']}/{network.prefixlen}"
+                    ],
                     "mtu": MTU,
                     "dhcp4": False,
                     "dhcp6": False,
                     "match": {"macaddress": mac_address},
+                    "routes": [
+                        {
+                            "to": str(network.network_address),
+                            "via": gateway,
```

Comment:
Is it worth us being explicit here with a 'metric': 100 since this is a default route and we can increment v2 route metric definitions in the future if we have additional subordinate routes that should be de-prioritized ?

Reply:
We can take metric setup and auto-increment on secondary routes to a followup effort on oracle secondary NIC configuration. I think there are a couple of inconsistencies with how secondary nics are setup in cloud-init vs Oracle's documented https://docs.oracle.com/en-us/iaas/Content/Resources/Assets/secondary_vnic_all_configure.sh